src/servers/src/elasticsearch.rs

```diff
@@ -261,49 +272,92 @@ fn convert_es_input_to_log_values(
         }
     );
 
-    let mut log_values: Vec<Value> = Vec::with_capacity(values.len() / 2);
+    let mut requests: Vec<LogIngestRequest> = Vec::with_capacity(values.len() / 2);
+    let mut values = values.into_iter();
 
+    // Read the ndjson payload and convert it to a (index, value) vector.
     // For Elasticsearch post `_bulk` API, each chunk contains two objects:
-    // 1. The first object is the command, it should be `create` or `index`. `create` is used for insert, `index` is used for upsert.
-    // 2. The second object is the document data.
-    let mut is_document = false;
-    for v in values {
-        if !is_document {
-            // Read the first object to get the command, it should be `create` or `index`.
-            ensure!(
-                v.get("create").is_some() || v.get("index").is_some(),
-                InvalidElasticsearchInputSnafu {
-                    reason: format!(
-                        "invalid bulk request, expected 'create' or 'index' but got {:?}",
-                        v
-                    ),
-                }
-            );
-            is_document = true;
-            continue;
+    //   1. The first object is the command, it should be `create` or `index`.
+    //   2. The second object is the document data.
+    loop {
+        // Read the first object to get the command, it should be `create` or `index`.
+        // Stop the loop if there is no command.
+        let next_cmd = values.next();
+        if next_cmd.is_none() {
+            break;
         }
+        let mut cmd = next_cmd.unwrap();
```

Comment:
We can use 

```rust
while let Some(cmd) = values.next() { ... }
```

to avoid use `unwrap` even it's safe

Reply:
TIL